### PR TITLE
play inline video on mobile CORE-8609

### PR DESCRIPTION
**Pull request by mmaxim**
@keybase/react-hackers 

This patch makes tapping a video in a chat thread start to play it inline in the thread on mobile, like on desktop. Each inline video is actually a `WebView` control that sources a webpage served by the service. Users can access the fullscreen option on iOS by using the control on the HTML5 viewer, but unfortunately Android users cannot (the fullscreen control doesn't work in the WebView there) (this is also how it works now on Android, it just launches a fullscreen view to play a small video). 

Note this patch includes a fork of the `react-native-wkwebview-reborn` library to allow us to play inline video in the `WKWebView` control on iOS. Let me know if I didn't set that up right.

### CODE REVIEW

**Comment by mmaxim** [comment]
This is the change on the webview library https://github.com/keybase/react-native-wkwebview/commit/8812e8cfec2a6959d1df0869b4112769f493c62e

File: shared/chat/conversation/messages/attachment/image/image-render.native.js [hunk 1]
```diff
@@ -1,11 +1,69 @@
 // @flow
 import * as React from 'react'
+import {WebView} from 'react-native'
+import WKWebView from 'react-native-wkwebview-reborn'
 import {NativeImage, NativeDimensions} from '../../../../../common-adapters/native-wrappers.native'
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/attachment/image/image-render.native.js]: WebView is exported as NativeWebView in native-wrappers.native. outside of common-adapters we shouldn't really import react-native directly

File: shared/chat/conversation/messages/attachment/image/image-render.native.js [hunk 2]
```diff
@@ -1,11 +1,69 @@
...
+  constructor(props: Props) {
+    super(props)
+    this.playingVideo = false
+  }
+
+  onVideoClick = () => {
+    if (!this.webview) {
+      return
+    }
+    const arg = !this.playingVideo ? 'play' : 'pause'
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/attachment/image/image-render.native.js]: prefer
`const arg = this.playingVideo ? 'pause' : 'play'`

File: shared/chat/conversation/messages/attachment/image/image-render.native.js [hunk 3]
```diff
@@ -1,11 +1,69 @@
...
+    const js = `togglePlay("${arg}")`
+    ;(isIOS ? this.webview.evaluateJavaScript : this.webview.injectJavaScript)(js)
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/attachment/image/image-render.native.js]: prettier is making this look strange, prefer
```
const runJS = isIOS ? this.webview.evaluateJavasscript : this.webview.injectJavascript
runJS(`togglePlay("${arg}")`)
```

File: shared/chat/conversation/messages/attachment/image/image-render.native.js [hunk 4]
```diff
@@ -1,11 +1,69 @@
...
+    this.playingVideo = !this.playingVideo
+  }
+
+  render() {
+    if (this.props.inlineVideoPlayable && this.props.videoSrc.length > 0) {
+      const uri = this.props.videoSrc + '&poster=' + encodeURIComponent(this.props.src)
+      const source = {uri}
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/attachment/image/image-render.native.js]: prefer
```
const source = {
  uri: `${this.props.videoSrc}&poster=${encodeURIComponent(this.props.src)}`
}
```

File: shared/chat/conversation/messages/attachment/image/image-render.native.js [hunk 5]
```diff
@@ -1,11 +1,69 @@
...
+      const allLoads = () => {
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/attachment/image/image-render.native.js]: should just be a method not made inline each time

File: shared/chat/conversation/messages/attachment/image/image-render.native.js [hunk 6]
```diff
@@ -1,11 +1,69 @@
...
+        this.props.onLoad()
+        this.props.onLoadedVideo()
+      }
+      return isIOS ? (
+        <WKWebView
+          ref={ref => {
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/attachment/image/image-render.native.js]: prefer createRef

File: shared/chat/conversation/messages/attachment/image/index.js [hunk 7]
```diff
@@ -102,7 +98,9 @@ class ImageAttachment extends React.PureComponent<Props, State> {
         >
           {!!this.props.path && (
             <ImageRender
-              ref={this.imageRef}
+              ref={ref => {
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/attachment/image/index.js]: why remove createRef?

Comment (mmaxim) [reply shared/chat/conversation/messages/attachment/image/index.js]: It wasn't working on native.

Comment (mmaxim) [reply shared/chat/conversation/messages/attachment/image/index.js]: Can't overwrite read-only property or something.

**Comment by mmaxim** [comment]
We talked about the last point in chat, and I'd like to use this on my trip so I'm going to merge. If there is anything else you want to change I can follow up.

**Pull request merged**